recipes/libtool/all/conanfile.py

```diff
@@ -26,106 +31,80 @@ class LibtoolConan(ConanFile):
+            self.win_bash = True
+            self.tool_requires("automake/1.16.5")     # Needed for complie and lib wrappers
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    @contextmanager
-    def _build_context(self):
-        with tools.run_environment(self):
-            with tools.environment_append(self._libtool_relocatable_env):
-                if self.settings.compiler == "Visual Studio":
-                    with tools.vcvars(self.settings):
-                        with tools.environment_append({"CC": "cl -nologo", "CXX": "cl -nologo",}):
-                            yield
-                else:
-                    yield
-
-    @property
-    def _datarootdir(self):
-        return os.path.join(self.package_folder, "res")
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) >= "12":
-            self._autotools.flags.append("-FS")
-        conf_args = [
-            "--datarootdir={}".format(tools.unix_path(self._datarootdir)),
-            "--prefix={}".format(tools.unix_path(self.package_folder)),
-            "--enable-shared",
-            "--enable-static",
-            "--enable-ltdl-install",
-        ]
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
-
-    @property
-    def _user_info_build(self):
-        return getattr(self, "user_info_build", self.deps_user_info)
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        if (self.settings.get_safe("compiler") == "Visual Studio" and self.settings.get_safe("compiler.version") >= "12") or \
+           (self.settings.get_safe("compiler") == "msvc" and self.settings.get_safe("compiler.version") >= "180"):
+            tc.extra_cflags.append("-FS")
+        tc.configure_args.append("--enable-shared")
+        tc.configure_args.append("--enable-static")
+        tc.configure_args.append("--enable-ltdl-install")
+        env = tc.environment()
+        if is_msvc(self):
+            env.append("CC", f'{unix_path(self, self.conf.get("user.automake:compile-wrapper"))} cl -nologo')
+            env.append("CXX", f'{unix_path(self, self.conf.get("user.automake:compile-wrapper"))} cl -nologo')
+            env.append("AR", f'{unix_path(self, self.conf.get("user.automake:lib-wrapper"))} lib')
+            env.append("LD", "link")
```

Comment:
If this is a fix for the specific issue that the libtool package fails to build on Windows via MSVC&mdash;which I'm currently hitting :)&mdash;would it be possible to create a separate PR for it?  It's just that this PR is currently failing the non-essential Conan 2.0 build checks. This type of failure looks like it would normally be non-blocking, but this PR is specifically about upgrading the package to work with Conan 2.0, so it seems like it could block the PR.

If this small change can get the package to work for Windows MSVC users, then it seems like a quick win to just merge it in isolation, without having to wait for the huge pile of changes in the current PR to be fixed for Conan 2.0 and merged. Right?

(Full disclosure: this is my first look at any PR on conan-io, so there could be a thousand good reasons why the above suggestion is laughably nonsensical. Apologies if that's the case!)

Reply:
Hi @kevinchannon  - absolutely! 
if I'm not mistaken, this is a specific fix to build on Windows with Visual Studio 2022 - the recipe as it is should still builds with earlier versions (without the need for this change). If that isn't the case, please let me know!

I'm aiming to have this PR reviewed and merged sometime tomorrow, including this fix, but otherwise feel free to open a PR just for the changes to this issue, referencing the upstream thread https://savannah.gnu.org/patch/?9313#comment1, so that we know that the workaround can be removed in a future version if they decide to accept the patch. I think these two lines disable fortran compiler detection